core/src/HIP/Kokkos_HIP_Parallel_Range.hpp

```diff
@@ -163,13 +177,14 @@ class ParallelReduce<CombinedFunctorReducerType, Kokkos::RangePolicy<Traits...>,
 
   __device__ inline void run(SHMEMReductionTag) const {
     const ReducerType& reducer = m_functor_reducer.get_reducer();
-    const integral_nonzero_constant<
-        size_type, ReducerType::static_value_size() / sizeof(size_type)>
+    const integral_nonzero_constant<word_size_type,
+                                    ReducerType::static_value_size() /
+                                        sizeof(word_size_type)>
         word_count(reducer.value_size() / sizeof(size_type));
```

Comment:
Shouldn't this be
```suggestion
        word_count(reducer.value_size() / sizeof(word_size_type));
```
?

Reply:
Ahh, yes, you're right.